### ISSUE TITLE
IOException thrown from MantaEncryptedObjectInputStream during read() missing context values

### ISSUE THREAD

**Issue by tjcelaya** [joyent/java-manta#260]
The following exception was reported, but no context was included. It looks like we rethrow caught exceptions without adding any context in

https://github.com/joyent/java-manta/blob/2a591afa0b501e008e69e1a7c762889eeb1ff548/java-manta-client/src/main/java/com/joyent/manta/client/crypto/MantaEncryptedObjectInputStream.java#L472

```
com.joyent.manta.org.apache.http.ConnectionClosedException: Premature end of Content-Length delimited message body (expected: 3366879; received: 2243367
        at com.joyent.manta.org.apache.http.impl.io.ContentLengthInputStream.read(ContentLengthInputStream.java:178) ~[validator.jar:?]
        at com.joyent.manta.org.apache.http.conn.EofSensorInputStream.read(EofSensorInputStream.java:135) ~[validator.jar:?]
        at com.joyent.manta.org.apache.commons.io.input.BoundedInputStream.read(BoundedInputStream.java:123) ~[validator.jar:?]
        at com.joyent.manta.org.apache.commons.io.input.BoundedInputStream.read(BoundedInputStream.java:105) ~[validator.jar:?]
        at org.bouncycastle.jcajce.io.CipherInputStream.nextChunk(Unknown Source) ~[bcprov-jdk15on-156.jar:1.56.0]
        at org.bouncycastle.jcajce.io.CipherInputStream.read(Unknown Source) ~[bcprov-jdk15on-156.jar:1.56.0]
        at java.io.FilterInputStream.read(FilterInputStream.java:107) ~[?:1.8.0_65]
        at com.joyent.manta.client.crypto.MantaEncryptedObjectInputStream.read(MantaEncryptedObjectInputStream.java:465) ~[validator.jar:?]
        at com.joyent.manta.client.crypto.MantaEncryptedObjectInputStream.read(MantaEncryptedObjectInputStream.java:415) ~[validator.jar:?]
        at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2146) ~[validator.jar:?]
        at org.apache.commons.io.IOUtils.copy(IOUtils.java:2102) ~[validator.jar:?]
        at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2123) ~[validator.jar:?]
```

It seems straightforward to at the very least replace the bare

```
else {
    throw e;
}
```
with
```
else {
    MantaIOException me = new MantaIOException(e);
    annotateException(me);
    throw me;
}
```

**Comment by cburroughs** [comment]
This is the issue addressed by #259 , correct?

**Comment by tjcelaya** [comment]
Yes I created the issue in parallel with @dekobon starting on that PR. My internet is slower!
